### ISSUE TITLE
Connects to channels before authenticating

### ISSUE THREAD

**Issue by collodrollo** [squidowl/halloy#229]
There are some channels that require authentication to join. #javascript on Libera, for example. These channels will not be joined on halloy startup because nickserv authentication has not occurred by the time attempts to join them have been made.

**Comment by avoidik** [comment]
you could set it as follows:

```
servers:
  libera:
    nickname: user
    nick_password: password 
```

**Comment by casperstorm** [comment]
You also have this: https://github.com/squidowl/halloy/wiki/Configuration#on-connect-commands

**Comment by casperstorm** [comment]
Feel free to reopen if you still have issues!

**Comment by o-alquimista** [comment]
This is supposed to be fixed by #346 `Handle ERR_NEEDREGGEDNICK (aka ERR_NOCHANMODES)`, right?
